tests/unit/apps/test_core.py

```diff
@@ -877,3 +877,101 @@ def _generate_nova_compute_app(model):
     )
 
     return app
+
+
+def test_cinder_upgrade_plan(model):
+    """Test generate plan for cinder application."""
+    target = OpenStackRelease("victoria")
+    exp_plan = dedent_plan(
+        """
+    Upgrade plan for 'cinder' to victoria
+        Upgrade software packages of 'cinder' from the current APT repositories
+            Upgrade software packages on unit cinder/0
+            Upgrade software packages on unit cinder/1
+            Upgrade software packages on unit cinder/2
+        Refresh 'cinder' to the latest revision of 'ussuri/stable'
+        Change 'action-managed-upgrade' config option to False.
+        Upgrade 'cinder' to the new channel: 'victoria/stable'
+        Change charm config of 'cinder' 'openstack-origin' to 'cloud:focal-victoria'
+        Wait 300s for app cinder to reach the idle state.
+        Check if the workload of 'cinder' has been upgraded on units: cinder/0, cinder/1, cinder/2
```

Comment:
you're missing running the upgrade action; if these are the steps, the workload check will fail as no upgrade will have occurred

Reply:
Yes, because I did not add any mores steps to Cinder, yet. Converting to draft + move fix for set `action-managed-upgrade` to #263

Reply:
I don't see the need of creating a cinder class right now. I think passing units to upgrade an OpenStackApplication should be able to run the pause, upgrade and resume actions of the units to upgrade.